### DISCUSSION TITLE
Have you considered the ubicloud rust-cache micro-fork?

### DISCUSSION THREAD

**Discussion by fdr** [ArroyoSystems/arroyo#758] [Ideas]
Ubicloud supports a more localized caching for rust artifacts.  I remember Arroyo was the first to assess that the imbalance between network locality and the increase in affordable processor performance meant it was faster to recompute a bunch of compilation artifacts, rather than cache. But, that probably is not true anymore.

The way this works is you can use the ubicloud micro-fork of the rust-cache github action: https://www.ubicloud.com/docs/github-actions-integration/ubicloud-cache

If you wait around long enough, though, @velioglu will likely ship transparent caching if we can fix all the remaining problems with it, then you would be able to use any upstream action without modification.

**Comment by mwylde** [comment]
Thanks Daniel for pointing that out! We'll give it a shot.
